README.md

```diff
@@ -100,11 +97,46 @@ package { 'PSExcel-psc':
 }
 ```
 
+### The provider
+
+The provider to use will either be `windowspowershell` or `powershellcore`. Nodes using `powershell.exe` will use `windowspowershell`, and nodes that have PowerShell core (`pwsh.exe`) will use the `powershellcore` provider with both the `psrepository` and `package` types.
+
 ## Limitations
 
 Note that PowerShell modules can be installed side by side so installing a newer
 version of a module will not remove any previous versions.
 
+## Reference
+
+### Types
+
+### psrepository
+
+Allows you to specify and configure a repository. The type expects a valid OneGet package provider source over an HTTP or HTTPS url.
```

Comment:
> ... valid OneGet package provider source...

This implies I can use this for any of those providers, including, say, the nano packages or chocolatey.